src/appleseed.shaders/include/appleseed/texture/as_texture_helpers.h

```diff
@@ -111,4 +111,43 @@ string get_interpolation_method(int method)
     }
 }
 
+color get_projection_color(
+    string filename,
+    float s_coord,
+    float t_coord,
+    float width,
+    float height,
+    float x_offset,
+    float y_offset,
+    float angle,
+    string swrap,
+    string twrap,
+    int sflip,
+    int tflip,
+    output float alpha)
+{
+    float st[2] = {s_coord, t_coord};
+
+    if (angle != 0.0)
+    {
+        rotate2d(st[0], st[1], angle, st[0], st[1]);
+    }
+
+    st[0] = sflip ? 1.0 - mod(st[0], UVWRAP) : mod(st[0], UVWRAP);
+    st[1] = tflip ? 1.0 - mod(st[1], UVWRAP) : mod(st[1], UVWRAP);
+
+    st[0] *= width + x_offset;
```

Comment:
Are you sure you want to multiply by `(width + x_offset)`?

Reply:
no, that is a bug, thanks :)